### PR TITLE
Decimal to binary conversion

### PR DESCRIPTION
**Pull request by wruoting**

### CODE REVIEW

File: core/src/main/scala/design/hamu/core/command/DecimalToBinaryCommand.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          isNegative = true
+          -decimal
+        }
+      }
+      var mutableDecimal = abs(decimal.toInt)
+      var mutableString = ""
+      if (mutableDecimal == 0) {
+        s"0"
+      } else {
+        while (mutableDecimal != 0) {
```
Comment (vangogh500) [inline core/src/main/scala/design/hamu/core/command/DecimalToBinaryCommand.scala]: So in FP we don't really use `loops` because that usually means side effects and mutability like what we are doing with `mutableDecimal` and `mutableString`. You can accomplish the same effect using recursion.

You can try doing it yourself for practice but I created a gist that is fully functional here. So you can take a look to reference later.
https://gist.github.com/vangogh500/e4b18e2e18629e244381193e24d835c0

File: core/src/main/scala/design/hamu/core/command/DecimalToBinaryCommand.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+object DecimalToBinaryCommand extends Command {
+
+  def run(ls: List[String]): String = ls match {
+    case decimal :: tail =>
+      var isNegative = false
+      val abs = (decimal: Int) => {
+        if (decimal >= 0) {
+          decimal
+        } else {
+          isNegative = true
```
Comment (vangogh500) [inline core/src/main/scala/design/hamu/core/command/DecimalToBinaryCommand.scala]: Another example of bad mutability/side effects :)
Also you can just use `Math.abs` here since I think `abs` is pretty trivial implementation.

File: core/src/test/scala/design/hamu/core/command/DecimalToBinaryCommandSpec.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package test.scala.design.hamu.core.command
+
+import design.hamu.core.command.DecimalToBinaryCommand
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+class DecimalToBinaryCommandSpec extends AnyWordSpec with Matchers {
+  "DecimalToBinary command" should {
+    "given a positive int" in {
+      DecimalToBinaryCommand.run(List("20")) must equal("10100")
```
Comment (vangogh500) [inline core/src/test/scala/design/hamu/core/command/DecimalToBinaryCommandSpec.scala:10]: Nice set of edge cases :+1:

**Review by vangogh500 — APPROVED** [review]
I'd say overall a good imperative implementation and seems like you got a good grasp of the language part.

Next step is to try to do things more functionally (no vars and no side effects). I've added a gist in one of the comments for a FP solution. We'll talk more about it today as well :+1:

**Comment by vangogh500** [comment]
Oh yea if you want to fix code formatting issue you are getting in CICD. Run `scalafmtAll` in sbt. It will auto format all your code.
